openpype/modules/kitsu/plugins/publish/collect_kitsu_entities.py

```diff
@@ -13,61 +17,67 @@ class CollectKitsuEntities(pyblish.api.ContextPlugin):
 
     def process(self, context):
 
-        asset_data = context.data["assetEntity"]["data"]
-        zou_asset_data = asset_data.get("zou")
+        # Get all needed names
+        project_name = context.data.get("projectName")
+        asset_name = None
+        task_name = None
+        # If asset and task name doesn't exist in context, look in instance
+        for instance in context:
+            asset_name = instance.data.get("asset")
+            task_name = instance.data.get("task")
```

Comment:
But you can have multiple instances in context and not all of them will have the same asset and task name. For example multi-shot workflow where you would publish in one go instances to multiple assets. Wouldn't this just use the asset and task names from the last instance in the context?

Reply:
Thanks for the great insight! It's my first PR working with `context` and `instance` so I didn't fully understand it until now. Fixing everything as we speak! :)